docs/pre-registration-developer-setup.md

```diff
@@ -0,0 +1,156 @@
+# Pre-registration Developer Setup
+
+## Overview
+The documentation here will guide you through the pre-requisites required for pre-registration developer setup. To know more, read [Pre-registration](https://docs.mosip.io/1.2.0/modules/pre-registration). 
+
+
+## Software setup
+
+### Below are the list of tools required in pre-registration:
+1. JDK 11
+2. Any IDE (Eclipse, IntelliJ IDEA)
+3. Apache Maven (zip folder)
+4. pgAdmin
+5. Postman
+6. Git
+7. [lombok.jar](https://projectlombok.org/download) (file)
+8. MOSIP pre-registration specific jar:
+   * [kernel-auth-adapter](https://repo1.maven.org/maven2/io/mosip/kernel/kernel-auth-adapter/) 
+   * [kernel-transliteration-icu4j](https://repo1.maven.org/maven2/io/mosip/kernel/kernel-transliteration-icu4j/)
+   * [kernel-ref-idobjectvalidator](https://repo1.maven.org/maven2/io/mosip/kernel/kernel-ref-idobjectvalidator/)
+   * [kernel-virusscanner-clamav](https://repo1.maven.org/maven2/io/mosip/kernel/kernel-virusscanner-clamav/)
+9. [settings.xml](https://github.com/aihamh/documentation/blob/1.2.0/docs/_files/pre-registration-config-files/settings.xml)
+10. Notepad++ (optional)
+
+
+### Follow the steps below to setup pre-registration on your local system:
+1. Fork the MOSIP [pre-registration repository](https://github.com/mosip/pre-registration) from Github [MOSIP repo](https://github.com/mosip).
+
+2. Clone the repo into local.
+   * `git clone https://github.com/urgithubaccname/pre-registration.git`
+   * `git checkout develop`
+   * `git remote add upstream https://github.com/mosip/pre-registration.git`
+   * `git remote set-url --push upstream no_push`
+   * `git remote -v`
+   * `git fetch upstream`
+   * `git rebase upstream/develop`
+ 	
+3. Inside `settings.xml` change local repository directory to your directory name where `.m2 folder` is located.
+	ex: `<localRepository>C:/Users/username/.m2/repository</localRepository>` found in 55th line
+
+4. Add `settings.xml` inside `.m2 folder` (Maven Folder).
+	ex: `C:\Users\username\.m2`
+
+5. Import the project in Eclipse IDE and it starts updating Maven projects configuration, Refreshing workspaces, project starts building (downloading sources, javadoc).
+
+6. Add Downloaded Lombok.jar to project, click on downloaded jar and install specifying eclipse ide(eclipse.exe) location.
+
+ <img src="_images/lombok-configuration.png" width="750" height="450">
+ 
+7. Configure the JDK (Standard VM) with your Eclipse by traversing through `Preferences → Java → Installed JREs`.
+
+<img src="_images/installed-jre.png" width="750" height="450">
+
+8. Add MOSIP pre-registration specific jar files from [Maven central](https://repo1.maven.org/maven2/io/mosip/):
+   * Adding Jars to build path: Right click on service -> Build Path -> Configure Build Path. click on Classpath, Add External Jars -> Add required Jars -> Apply and close.
```

Comment:
* Adding Jars to build path: Right click on the service -> Build Path -> Configure Build Path -> Click on Classpath -> Add External Jars -> Add required Jars -> Apply and Close.